ginkgo: fix test falures

06kellyjac opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Changes in #161985 broke ginkgo's tests since it doesn't
respect excludedPackages in checks anymore

Why did #161985 not put `grep -q "$exclude" <<<$pkg && continue` within the `buildGoDir` function to retain the previous behavior?

Fixes a hydra failure: https://hydra.nixos.org/build/173572510

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

06kellyjac commented:
Result of `nixpkgs-review pr 168493` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ginkgo</li>
  </ul>
</details>

saschagrunert reviewed (approved):
LGTM

mmlb commented:
This is already fixed in staging: https://github.com/NixOS/nixpkgs/pull/167912

mmlb commented:
> ###### Description of changes
> 
> Changes in #161985 broke ginkgo's tests since it doesn't respect excludedPackages in checks anymore
> 
> Why did #161985 not put `grep -q "$exclude" <<<$pkg && continue` within the `buildGoDir` function to retain the previous behavior?

The reason I started in on #161985 was because the exclusion *was* happening in `buildGoDir` and there were confusing messages of "Building package" of a package that should have been excluded. #167912 puts the filtering into `getGoDirs` were it should have been always imo.

06kellyjac commented:
ah awesome :rocket:. I'll add a note that this can be removed when `#167912` hits master

---

While we're on the topic of `buildGoModule`, I still have a couple long standing issues with it

`$checkFlags` passed in to buildGoDir in the check phase actually breaks everything if it's actually set in nix :sweat_smile:
Also `subPackages` is super useful for when I want to build just 1 thing e.g. `cmd/somedir` and ignore all the other utility tools in the repo & you don't need to keep it updated constantly compared to excluding, but `subPackages` also limits the tests and means a bunch of go packages run no tests

06kellyjac committed:
ginkgo: fix test falures

Changes in #161985 broke ginkgo's tests since it doesn't
respect excludedPackages in checks anymore

Should be resolved by #167912 once it hits master

06kellyjac commented:
Result of `nixpkgs-review pr 168493` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ginkgo</li>
  </ul>
</details>

mmlb commented:
@06kellyjac code from #167912 is now on master as https://github.com/NixOS/nixpkgs/commit/ddcc7c67d280eb0ea76a9079c49c9f70c6cc2e4f. I've verified that ginkgo builds and tests pass as of a fresh pull of master ( c52e644aeaec6e036e79b34e231275f3c8090913) so this can just be closed.

06kellyjac commented:
Awesome, thanks @mmlb
